core/src/subgraph/instance_manager.rs

```diff
@@ -639,221 +617,195 @@ where
 
     // Process events one after the other, passing in entity operations
     // collected previously to every new event being processed
-    process_triggers(
-        logger.clone(),
+    let (mut ctx, mut block_state) = process_triggers(
+        &logger,
         BlockState::with_cache(std::mem::take(&mut ctx.state.entity_lfu_cache)),
         ctx,
```

Comment:
Hmm, are `&mut ctx` and `&mut block_state` possible here? I see more opportunities for breaking with that functional approach that was necessary before async/await.